Add more test cases

geoffbacon opened the pull request:
Hi fievelk,

I've added some more test cases to test_aline.py. What do you think?

ExplodingCabbage committed:
Fix more nltk.six imports

I get ImportErrors when trying to run tox otherwise; as far as I can tell, this case is the same as eebeab6?

fievelk committed:
Merge pull request #1 from ExplodingCabbage/fix/six-import

Fix more nltk.six imports

fievelk committed:
Refactor import statements

fievelk committed:
Add pyparsing to tox dependencies

Fix NameError: global name 'pyparsing' is not defined

fievelk committed:
Skip Senna doctests and use Unittest instead

fievelk committed:
Add python-crfsuite to tox dependencies

fievelk committed:
Make tox aware of environment variables

For example, this fix allows tox to find NLTK data when a custom NLTK_DATA environment variable is set

fievelk committed:
Add verbose flag to TransitionParser training method

This allows to avoid unnecessarily verbose output in tests

fievelk committed:
Remove TransitionParser temporary files after tests

Remove related entries from gitignore

fievelk committed:
Skip senna unittests when executable is not found

fievelk committed:
Refactor aline.py

- Remove EOL whitespaces
- Fix indentation

fievelk committed:
Fix aline tests

Use unittest instead of doctest and import unicode_literals

geoffbacon committed:
Add more test cases

fievelk commented:
@geoffbacon Thank you for your PR.

Are these assertions always testing the same concept? If they all test the same functionality, then we can keep them as they are (even if I think it would be better to have 3 or 4 at max).

If, on the contrary, each assertion is testing a different concept I would rather create other test methods with explanatory names for each one of them.

What do you think?

geoffbacon committed:
add test of delta method

geoffbacon committed:
whitespace cleanup

geoffbacon commented:
They're testing the same concept (although that concept is the entire functionality of the module), so I've removed all but four and added another test for a different concept (the delta function). The new test for the delta function will most likely fail if any of the constants at the top of the file are changed (C_skip, similarity_matrix, or salience) but these should never change as their values are from the original work (Kondrak 2002).

fievelk commented:
Thank you @geoffbacon. I cherry-picked your updates and added them to the `fievelk/test-fixes` branch. They are now part of nltk/nltk#1485 :)
